WPMedia/engine-theme-sdk
# TMEDIA-962 Removed unused functionality for clarity, performance

## Jira Ticket

- [TMEDIA-963](https://arcpublishing.atlassian.net/browse/TMEDIA-963)

## Acceptance Criteria

Media and commerce pages from current environments should run without new errors compared to dev environments 

Measure difference in bundle size (tree-shaking?); couldn't reliably test with the minifest false on. Compile would at least be faster with installing fewer dependencies here

All functionality except for designated ones in checklist engine are included `SDK will still use MetaData and LazyLoad` [Converting a v1 block to v2 architecture](https://arcpublishing.atlassian.net/wiki/spaces/TI/pages/3106243821/Converting+a+v1+block+to+v2+architecture)

## Test Steps

- Link engine-theme-sdk in the commerce. Run `npx fusion start -f` to ensure all pages in data download for dev arc commerce work as expected -> all look good here 
<img width="710" alt="Screen Shot 2022-10-03 at 14 38 03" src="https://user-images.githubusercontent.com/5950956/193663879-f6fd66f3-c980-4f91-a841-4a19663cab55.png">

- Link engine-theme-sdk in news. Run `npx fusion start -f` to ensure all pages in data download for dev themes internal work as expected -> working with the data from themes internal

## Effect Of Changes

### Before

- Possibility of using unsupported code in engine theme sdk 

### After

- Less code to compile and fewer dependencies to install to compile down. 

## Dependencies or Side Effects

- Blocks in v2 aren't using these any way

## Review Checklist

        please explain why (so that we can fix it whenever it gets refactored).

# Test 


---- 

- Remove components not needed in v2 to prep for that 
- Also see the performance gains and what we need to do to fully migrate

Removed: 
- Video 
- VideoPlayer 
- Image 
- ErrorBoundary 
- Gallery 
- Icons
- Lightbox

^ associated packages of above 

todo: 
- Looks like we're still relying upon Metadata in the Output types. This is using the v1 resizer logic. Can't get rid of thumbor lite yet
- Test locally with v2 commerce and news
